Resources/Computing/SSHComputingElement.py

```diff
@@ -505,28 +510,44 @@ def submitJob(self, executableFile, proxy, numberOfJobs=1):
     else:  # no proxy
       submitFile = executableFile
 
-    result = self._submitJobToHost(submitFile, numberOfJobs)
-    if proxy:
+    inputs = None
+    if self.parallelLibrary:
+      # In this case, the executable becomes a dependency of a parallel library script
+      inputs = [submitFile]
+      submitFile = self.parallelLib.generateWrapper(submitFile)
+
+    result = self._submitJobToHost(submitFile, numberOfJobs, inputs=inputs)
+    if proxy or self.parallelLibrary:
       os.remove(submitFile)
+      for inputFile in inputs:
+        os.remove(inputFile)
 
     return result
 
-  def _submitJobToHost(self, executableFile, numberOfJobs, host=None):
+  def _submitJobToHost(self, executableFile, numberOfJobs, host=None, inputs=None):
     """  Submit prepared executable to the given host
     """
     ssh = SSH(host=host, parameters=self.ceParameters)
     # Copy the executable
-    submitFile = '%s/%s' % (self.executableArea, os.path.basename(executableFile))
+    submitFile = os.path.join(self.executableArea, os.path.basename(executableFile))
     result = ssh.scpCall(30, executableFile, submitFile, postUploadCommand='chmod +x %s' % submitFile)
     if not result['OK']:
       return result
 
+    # Copy the executable dependencies if any
+    for localInput in inputs:
+      remoteInput = os.path.join(self.executableArea, os.path.basename(localInput))
+      result = ssh.scpCall(30, localInput, remoteInput, postUploadCommand='chmod +x %s' % remoteInput)
```

Comment:
I guess `remoteInput` should be quoted

Reply:
Why do you thinkg `remoteInput` should be quoted exactly?

Thanks a lot for all your comments @chrisburr

Reply:
The most common thing would be spaces in the path:
```bash
$ touch "/tmp/cburr/i am executable"
$ chmod +x /tmp/cburr/i am executable
chmod: cannot access ‘/tmp/cburr/i’: No such file or directory
chmod: cannot access ‘am’: No such file or directory
chmod: cannot access ‘executable’: No such file or directory
```
but it could be security issue if `remoteInput` is untrusted e.g.: `remoteInput="/tmp; sudo rm -rf /"`.

Reply:
It's generally better to use interfaces that accept lists like `subprocess.run(["chmod", "-x", "--", remoteInput])` but it's not always an option.

Note that even this is easy to get wrong as `"--"` is needed to ensure `chmod` doesn't get confused by paths starting with `-`, e.g.

```bash
$ touch -- --help
$ ls
--help
$ chmod +x --help
# contents of chmod --help
```

Again it can become a security issue if the string is untrusted.

Reply:
Alright, I added `shlex_quote` directly within the `scpCall()` method to wrap the `localFile` and the `remoteFile`.